sklearn/model_selection/_classification_threshold.py

```diff
@@ -0,0 +1,783 @@
+    ...     f"{classifier_tuned.objective_score_[1]:.3f}."
+    ... )
+    Cut-off point found at 0.3... for a recall of 0.7... and a precision of 0.7...
+    >>> print(classification_report(y_test, classifier_tuned.predict(X_test)))
+                  precision    recall  f1-score   support
+    <BLANKLINE>
+               0       0.96      0.96      0.96       224
+               1       0.68      0.65      0.67        26
+    <BLANKLINE>
+        accuracy                           0.93       250
+       macro avg       0.82      0.81      0.81       250
+    weighted avg       0.93      0.93      0.93       250
+    <BLANKLINE>
+    """
+
+    _parameter_constraints: dict = {
+        "estimator": [
+            HasMethods(["fit", "predict_proba"]),
+            HasMethods(["fit", "decision_function"]),
+        ],
+        "strategy": [StrOptions({"optimum", "constant"})],
+        "objective_metric": [
+            StrOptions(
+                set(get_scorer_names())
+                | {
+                    "max_tnr_at_tpr_constraint",
+                    "max_tpr_at_tnr_constraint",
+                    "max_precision_at_recall_constraint",
+                    "max_recall_at_precision_constraint",
+                }
+            ),
+            callable,
+            MutableMapping,
+        ],
+        "constraint_value": [Real, None],
+        "constant_threshold": [Real],
+        "pos_label": [Real, str, "boolean", None],
+        "response_method": [StrOptions({"auto", "predict_proba", "decision_function"})],
+        "n_thresholds": [Interval(Integral, 1, None, closed="left")],
+        "cv": [
+            "cv_object",
+            StrOptions({"prefit"}),
+            Interval(RealNotInt, 0.0, 1.0, closed="right"),
+        ],
+        "refit": ["boolean", StrOptions({"auto"})],
+        "n_jobs": [Integral, None],
+        "random_state": ["random_state"],
+    }
+
+    def __init__(
+        self,
+        estimator,
+        *,
+        strategy="optimum",
+        objective_metric="balanced_accuracy",
+        constraint_value=None,
+        constant_threshold=0.5,
+        pos_label=None,
+        response_method="auto",
+        n_thresholds=100,
+        cv=None,
+        refit="auto",
+        n_jobs=None,
+        random_state=None,
+    ):
+        self.estimator = estimator
+        self.strategy = strategy
+        self.objective_metric = objective_metric
+        self.constraint_value = constraint_value
+        self.constant_threshold = constant_threshold
+        self.pos_label = pos_label
+        self.response_method = response_method
+        self.n_thresholds = n_thresholds
+        self.cv = cv
+        self.refit = refit
+        self.n_jobs = n_jobs
+        self.random_state = random_state
+
+    @_fit_context(
+        # estimators in TunedThresholdClassifier.estimator is not validated yet
+        prefer_skip_nested_validation=True
+    )
+    def fit(self, X, y, **params):
```

Comment:
Logistically, do you think this PR will be smaller and easier to review if `**params` was removed from this PR and added as a follow up PR?

Without `**params`, reviewers do not need to consider routing yet. For me, I think there is already a benefit of releasing `TunedThresholdClassifier` without a `**params` at first.